pangeo_forge/recipe.py

```diff
@@ -197,37 +215,44 @@ def _prepare_target():
                     pass
 
             except (FileNotFoundError, IOError, zarr.errors.GroupNotFoundError):
-
-                init_dsets = []
+                logger.info("Creating a new dataset in target")
+                # need to rewrite this as an append loop
                 for chunk_key in self._init_chunks:
-                    chunk_ds = self.open_chunk(
-                        chunk_key
-                    ).chunk()  # make sure data are not in memory
-                    init_dsets.append(chunk_ds)
-                # TODO: create csutomizable option for this step
-                # How to combine attrs is particularly important. It seems like
-                # xarray is missing a "minimal" option to only keep the attrs
-                # that are the same among all input variables.
-                ds = xr.merge(
-                    init_dsets, compat="identical", join="exact", combine_attrs="override"
-                )
-
-                # make sure the concat dim has a valid fill_value to avoid
-                # overruns when writing chunk
-                # this is problematic because this value will always come out as NaN
-                # when we open up the target
-                # ds[self.sequence_dim].encoding = {"_FillValue": -1}
-                # actually not necessary if we use decode_times=False
-
-                if self.target_chunks:
-                    # target_chunks has undergone some validation in __post_init__
-                    ds = ds.chunk(self.target_chunks)
-
-                self.initialize_target(ds)
+                    with self.open_chunk(chunk_key) as ds:
+                        # need to have the data in memory to avoid weird chunk problems
+                        ds.load()
+
+                        # https://github.com/pydata/xarray/blob/5287c7b2546fc8848f539bb5ee66bb8d91d8496f/xarray/core/variable.py#L1069
+                        for v in ds.variables:
+                            if self.target_chunks:
+                                this_var = ds[v]
+                                chunks = {
+                                    this_var.get_axis_num(dim): chunk
+                                    for dim, chunk in self.target_chunks.items()
+                                    if dim in this_var.dims
+                                }
+
+                                chunks = tuple(
+                                    chunks.get(n, s) for n, s in enumerate(this_var.shape)
+                                )
+                                ds[v].encoding["chunks"] = chunks
+                            else:
+                                ds[v].encoding["chunks"] = ds[v].shape
+
+                        target_mapper = self.target.get_mapper()
+                        ds.to_zarr(target_mapper, mode="a", compute=False)
```

Comment:
`compute=False` does not have the desired effect if the data are not dask arrays.

Reply:
Here (and or maybe up above before `load`) are you able to slice all the dimensions to size 0? Does that accomplish essentially the same thing?

```
ds = ds.isel(**{k: slice(0) for k in ds.dims})
```

Reply:
But then the target variables wouldn't get the right dimensions.

I think https://github.com/pydata/xarray/issues/5056 could help. It would allow us to call `chunk` on the inputs but not worry about potential conflicts with the target chunks.